serve.js

```diff
@@ -110,38 +107,63 @@ function serve(config, cb) {
 function initTasks(cb) {
   log.info('Initializing tasks');
 
-  _.each(Object.keys(config.tasks),function(name) {
-
-    var pattern = config.tasks[name]
-    ,  rule = new schedule.RecurrenceRule();
+  // first kick off the tasks
+  async.eachSeries(Object.keys(config.tasks), function(name,done) {
+    triggerTask(name,done);
+    done()
+  }, function(err) {
 
-    rule.minute = new schedule.Range(0, 59, pattern.minute);
+    if(err) log.err("Error initializing tasks",err);
 
-    schedule.scheduleJob(rule, function(name) {
+    //then set the intervals
+    _.each(Object.keys(config.tasks),function(name,i) {
 
-      var cdn = name
-        , task = require('./tasks/task')
-        , scrape = null;
+      var pattern = config.tasks[name];
 
-      log.info("running task...",name,pattern);
+      // we want to space out the syncs by 3 minutes each
+      setInterval(function(name) {
 
-      try
-      {
-        scrape = require('./tasks/' + cdn)(github);
-        task(config.output, cdn, scrape)(function (err) {
-          if (err)
-            log.err(err);
-        });
-      } catch(e) {
-        log.err(e);
-      }
-    }.bind(null,name));
+        triggerTask(name);
+      }.bind(null,name),6*(i*1e4 + pattern.minute*1e4));
+    });
   });
 }
 
+function triggerTask(name,done) {
+
```

Comment:
Morally opposed to promises (not totally), if we're going to accept promises as the standard coding style for the project I'll conform - though generally I prefer not to. I'm not using an async pattern as that function must be called from a scheduled job (on an interval).